### ISSUE TITLE
Are all keypoints predicted with semantic label?

### ISSUE THREAD

**Issue by FlyleafPaopao** [Gabriel-SGama/Semantic-SuperPoint#1]
There are too few visulaized results in this repo. I have read the paper. Does this model only improve the keypoints' quality? Or this model predict the keypoints' label?

**Comment by Gabriel-SGama** [comment]
The model improves the quality of the keypoints and descriptors (not sure if you are using keypoint=feature=(keypoint + descriptor)) by being trained to also predict the semantic label. The initial idea was to also use the semantic prediction but the predicted semantic label wasn't helpful in any application.

I mentioned the possible causes here:
III. SEMANTIC SUPERPOINT MODEL
...
Due to the aggressive data augmentation process and the large variety of classes in the MS-COCO dataset [17], used for training, the semantic prediction is not usable in practical application. Nonetheless, adding a semantic decoder for training improved the quality of the deep feature extractor. It can therefore be said that rather than being a model, SSp is more like a training method.

In short, I predict the keypoint label but I don't use it. If this didn't answer your question, feel free to open another issue.

**Comment by FlyleafPaopao** [comment]
Thank you. I understand the idea. What I want to do is to train my own model. And I want that the keypoints are totally the semantic points, like facial points. Could I achieve this target based SSp?

**Comment by Gabriel-SGama** [comment]
By using the current model, probably not. The semantic prediction resolution is 1/8 of the original so that won't give the precision that you need.

Another concern is how good the semantic prediction would be. To solve this, you can change the model and/or the method of learning. The one used here is highly dependent on data augmentation, which hurts semantic learning.

By doing this, you could predict the keypoint and semantic segmentation to filter or classify the keypoints. So I think the idea behind the SSp could make what you are trying to do possible, but it needs some improvements on the training pipeline.

Everything I said here is based only on speculation about what you are trying to do so it might be wrong.

**Comment by FlyleafPaopao** [comment]
Thanks, I will try it.
